### PR TITLE
Reduces use of general exceptions

### PR DESCRIPTION
**Pull request by reinhapa**
prepares pending spotbugs/spotless merges #1108 and #1107 failing due  to usage of generic exception types

Signed-off-by: Patrick Reinhart <patrick@reini.net>

### CODE REVIEW

**Review by mklaehn — DISMISSED** [review]

File: spotbugs-exclude.xml [hunk 1]
```diff
@@ -23,6 +23,15 @@
 -->
 <FindBugsFilter>
     <!-- contains specific exclusions of bug patterns -->
+    <Match>
+        <Or>
+            <Class name="org.tweetwallfx.cache.URLContentCacheBase$1" />
+            <Class name="org.tweetwallfx.cache.URLContentCacheBase$2" />
+            <Class name="org.tweetwallfx.config.ConfigurationTest" />
```
Comment (mklaehn) [inline spotbugs-exclude.xml:30]: This entry is not necessary when you remove the throws declaration from the test method. That throws is unnecessary

File: spotbugs-exclude.xml [hunk 2]
```diff
@@ -23,6 +23,15 @@
...
+            <Class name="org.tweetwallfx.stepengine.api.StepEngine" />
```
Comment (mklaehn) [inline spotbugs-exclude.xml:31]: This entry could be removed if the caught and rethrown RuntimeException is wrapped in a e.g. IllegalStateException

File: util/src/main/java/org/tweetwallfx/util/testcase/RunnableTestCase.java [hunk 3]
```diff
@@ -33,5 +33,5 @@ public interface RunnableTestCase {
      *
      * @throws Exception an exception that might be caused in the test casae
```
Comment (mklaehn) [inline util/src/main/java/org/tweetwallfx/util/testcase/RunnableTestCase.java:34]:
```suggestion
```

File: spotbugs-exclude.xml (hunk 1 above)
Comment (mklaehn) [reply spotbugs-exclude.xml:30]: Also something you already did. So should be oky to remove this entry
```suggestion
```

File: util/src/main/java/org/tweetwallfx/util/testcase/RunnableTestCase.java (hunk 3 above)
Comment (reinhapa) [reply util/src/main/java/org/tweetwallfx/util/testcase/RunnableTestCase.java:34]: done

File: spotbugs-exclude.xml (hunk 2 above)
Comment (reinhapa) [reply spotbugs-exclude.xml:30]: The remaining exclusions will be needed as implemented interfaces declare `Exceptions` within the `javafx.concurrent.Task::call` and the catch block for `RuntimeException` within the `StepEngine`

**Review by mklaehn — APPROVED** [review]
Thank you

**Review by mklaehn — APPROVED** [review]
LGTM

**Pull request merged**